### PR TITLE
pbs_snapshot: adding ability to obfuscate existing snapshots

### PR DESCRIPTION
**Pull request by agrawalravi90**
#### Describe Bug or Feature

Often times, it’s useful to be able to obfuscate an existing snapshot. For instance: user took a snapshot of a hard to reproduce issue but forgot to run pbs_snapshot in obfuscation mode. Since the issue is hard to reproduce, it’d be much easier if user could just obfuscate the already captured snapshot. So, this will add the ability to pbs_snapshot to be able to take an existing snapshot and obfuscate it.

#### Describe Your Change

See design doc for details

#### Link to Design Doc

https://openpbs.atlassian.net/wiki/spaces/PD/pages/2732032016/pbs+snapshot+Add+the+ability+to+obfuscate+existing+snapshots

#### Attach Test and Valgrind Logs/Output

[test_obfuscate_existing.log](https://github.com/openpbs/openpbs/files/6328185/test_obfuscate_existing.log)

### CODE REVIEW

File: test/fw/bin/pbs_snapshot [hunk 1]
```diff
@@ -100,11 +101,57 @@ Usage: pbs_snapshot -o <path to existing output directory> [OPTION]
...
+    tar = tarfile.open(tarpath)
+    main_snap = tar.getnames()[0].split(os.sep, 1)[0]
+    main_snap = os.path.join(os.path.abspath(parentdir), main_snap)
+    if os.path.isdir(main_snap):
+        logger.error("Existing snapshot %s found, cannot extract tar" %
+                     main_snap)
+        tar.close()
+        return (main_snap, False)
+    tar.extractall(path=parentdir)
+    tar.close()
```
Comment (bhroam) [inline test/fw/bin/pbs_snapshot:151]: You used with tar.open() above, why not here?

**Review by bhroam — CHANGES_REQUESTED** [review]

**Review by bhroam — APPROVED** [review]
LGTM

**Comment by bhroam** [comment]
@agrawalravi90 CI failed due to a pycodestyle error.

**Review by ashwathraop — APPROVED** [review]

**Comment by agrawalravi90** [comment]
@bhroam thanks, I've fixed the pycodestyle issues, please take a look
@ashwathraop thanks for the sign-off!

**Pull request merged**